src/Symfony/Bundle/FrameworkBundle/Command/DebugAutoconfigurationCommand.php

```diff
@@ -0,0 +1,171 @@
+            ])
+            ->setDescription('Displays current autoconfiguration for an application')
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> command displays all services that
+are autoconfigured:
+
+  <info>php %command.full_name%</info>
+
+You can also pass a search term to filter the list:
+
+  <info>php %command.full_name% log</info>
+
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+        $errorIo = $io->getErrorStyle();
+
+        $autoconfiguredInstanceofItems = $this->getContainerBuilder()->getAutoconfiguredInstanceof();
+
+        if ($search = $input->getArgument('search')) {
+            $autoconfiguredInstanceofItems = array_filter($autoconfiguredInstanceofItems, function ($key) use ($search) {
+                return false !== stripos(str_replace('\\', '', $key), $search);
+            }, ARRAY_FILTER_USE_KEY);
+
+            if (empty($autoconfiguredInstanceofItems)) {
+                $errorIo->error(sprintf('No autoconfiguration interface/class found matching "%s"', $search));
+
+                return 1;
+            }
+        }
+
+        ksort($autoconfiguredInstanceofItems, SORT_NATURAL);
+
+        $io->title('Autoconfiguration');
+        if ($search) {
+            $io->text(sprintf('(only showing classes/interfaces matching <comment>%s</comment>)', $search));
+        }
+        $io->newLine();
+
+        /** @var ChildDefinition $autoconfiguredInstanceofItem */
+        foreach ($autoconfiguredInstanceofItems as $key => $autoconfiguredInstanceofItem) {
+            $tableRows = [];
+
+            foreach ($autoconfiguredInstanceofItem->getTags() as $tag => $tagAttributes) {
+                $tableRows[] = ['Tag', $tag];
+                if ($tagAttributes !== [[]]) {
+                    $tableRows[] = ['Tag attribute', $this->dumpTagAttribute($tagAttributes)];
+                }
+            }
+
+            if ($autoconfiguredInstanceofItem->getMethodCalls()) {
+                $tableRows[] = ['Method call', $this->dumpMethodCall($autoconfiguredInstanceofItem)];
+            }
+
+            if ($autoconfiguredInstanceofItem->getBindings()) {
+                $tableRows[] = ['Bindings', $this->dumpBindings($autoconfiguredInstanceofItem)];
+            }
+
+            $io->writeln(sprintf("Autoconfiguration for \"%s\"\n==============================================", $key));
+            $io->newLine();
+            $io->table(['Option', 'Value'], $tableRows);
+        }
+    }
+
+    private function dumpMethodCall(ChildDefinition $autoconfiguredInstanceofItem)
+    {
+        $tagContainerBuilder = new ContainerBuilder();
+        foreach ($tagContainerBuilder->getServiceIds() as $serviceId) {
+            $tagContainerBuilder->removeDefinition($serviceId);
```

Comment:
isnt $tagContainerBuilder empty already :thinking:

Reply:
I think it's to make sure. And we could refactor it to be save empty and clone it on each call, no ?